src/services/postServices.ts

```diff
@@ -0,0 +1,90 @@
+import { Document } from "mongoose";
+import { BadRequestError, ServerError } from "../errors/customErrors.ts";
+import PostModel, { Post } from "../models/Post.ts";
+import UserModel from "../models/UserModel.ts";
+
+export const getSinglePostDocument = async (
+  postId: string,
+): Promise<Document> => {
+  try {
+    const post = await PostModel.findById(postId);
+    if (!post) {
+      throw new ServerError("Post not found");
+    }
+    return post.toJSON() as Document;
+  } catch (err) {
+    throw new ServerError(`${err}`);
+  }
+};
+
+export const buildPostDocument = async (postBody: Post): Promise<string> => {
+  try {
+    const { author, header, body, media, privacy } = postBody;
+    const getUser = await UserModel.findById(author);
+
+    if (!getUser) {
+      throw new BadRequestError("Invalid author");
+    }
+    if (!body || !header) {
+      throw new BadRequestError("Missing information");
+    }
+    const updateData = {
+      author: author,
+      header: header,
+      body: body,
+      media: media,
+      privacy: privacy,
+      location: postBody.location,
+    };
+    const newPost = await PostModel.create(updateData);
+    if (!newPost) {
+      throw new ServerError("Failed to create post");
+    }
+    return newPost._id as string;
+  } catch (error) {
+    throw new ServerError(`${error}`);
+  }
+};
+export const updatePostDocument = async (
```

Comment:
> Please also check if the current user is the same as the author of the post to be able to update the post. However, it would be convenient if you create a middleware to handle this before the request reach this controller. you could do it in the post Router.

Hi Hung I just added the new middleware at [#ff32b3e](https://github.com/hungngodev/Flavorie/pull/82/commits/ff32b3e63f0d4646e2e0374df393137c31ab304f)!